### PR TITLE
fix: registry returned by getRegistryRepository must be a URL

### PR DESCRIPTION
**Pull request by candrews**
## Changes:
The registry returned by `getRegistryRepository` must be a URL (including the scheme portion).

## Context:

Addresses another issue covered in #8649 - specifically https://github.com/renovatebot/renovate/issues/8649#issuecomment-787794610

I'm seeing this stack trace:
```
"stack": "TypeError [ERR_INVALID_URL]: Invalid URL: my.registry/v2/
    at onParseError (internal/url.js:259:9)
    at parse (<anonymous>)
    at new URL (internal/url.js:335:5)
    at Object.exports.default (/usr/src/app/node_modules/got/dist/source/core/utils/options-to-url.js:35:17)
    at normalizeArguments (/usr/src/app/node_modules/got/dist/source/core/index.js:480:51)
    at Function.got [as default] (/usr/src/app/node_modules/got/dist/source/create.js:112:39)
    at gotRoutine (/usr/src/app/node_modules/renovate/lib/util/http/index.ts:79:25)
    at queueTask (/usr/src/app/node_modules/renovate/lib/util/http/index.ts:138:16)
    at Http.request (/usr/src/app/node_modules/renovate/lib/util/http/index.ts:145:45)
    at Http.get (/usr/src/app/node_modules/renovate/lib/util/http/index.ts:165:17)
    at getAuthHeaders (/usr/src/app/node_modules/renovate/lib/datasource/docker/index.ts:139:41)
    at getTags (/usr/src/app/node_modules/renovate/lib/datasource/docker/index.ts:445:27)
    at Object.getReleases (/usr/src/app/node_modules/renovate/lib/datasource/docker/index.ts:624:16)
    at getRegistryReleases (/usr/src/app/node_modules/renovate/lib/datasource/index.ts:68:15)
    at fetchReleases (/usr/src/app/node_modules/renovate/lib/datasource/index.ts:211:15)
    at Object.getPkgReleases (/usr/src/app/node_modules/renovate/lib/datasource/index.ts:272:7)
    at Object.lookupUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/lookup/index.ts:55:30)
    at fetchDepUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/fetch.ts:42:26)
    at /usr/src/app/node_modules/p-map/index.js:57:22"
```
`my.registry/v2/` is an invalid URL - the correct value is `https://my.registry/v2/`

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [X] Code inspection only, or
- [X] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by candrews** [comment]
@rarkins since you were the person who last worked in this area, can you please review (and hopefully merge) this PR?

**Review by rarkins — DISMISSED** [review]

File: lib/datasource/docker/index.ts [hunk 1]
```diff
@@ -66,9 +66,14 @@ export function getRegistryRepository(
   registryUrl: string
 ): RegistryRepository {
   if (registryUrl !== defaultRegistryUrls[0]) {
-    const registry = registryUrl.replace('https://', '');
-    const registryEndingWithSlash = registry.replace(/\/?$/, '/');
+    const registryEndingWithSlash = registryUrl
+      .replace(/^https?:\/\//, '')
+      .replace(/\/?$/, '/');
```
Comment (viceice) [inline lib/datasource/docker/index.ts:71]:
```suggestion
    const registryEndingWithSlash = ensureTrailingSlash(registryUrl.replace(/^https?:\/\//, ''));
```

It's in `util/url`. maybe needs running prettier.

File: lib/datasource/docker/index.ts [hunk 2]
```diff
@@ -66,9 +66,14 @@ export function getRegistryRepository(
...
     if (lookupName.startsWith(registryEndingWithSlash)) {
+      let registry = registryUrl.replace(/\/?$/, '');
```
Comment (viceice) [inline lib/datasource/docker/index.ts:73]: Maybe we should add a `trimTrailingSlash` util function too?

Comment (candrews) [reply lib/datasource/docker/index.ts:73]: Added :+1:

File: lib/datasource/docker/index.ts (hunk 1 above)
Comment (candrews) [reply lib/datasource/docker/index.ts]: Thanks for the tip - I've made that change. :+1:

**Comment by viceice** [comment]
@candrews Please do not force-push, simply merge master. We will squash the pr anyways. If you force-push we need to do a full review, because we don't see that's changed in latest commits

File: lib/datasource/docker/index.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ import { ExternalHostError } from '../../types/errors/external-host-error';
 import * as packageCache from '../../util/cache/package';
 import * as hostRules from '../../util/host-rules';
 import { Http, HttpResponse } from '../../util/http';
+import { ensureTrailingSlash, trimTrailingSlash } from '../util/url';
```
Comment (viceice) [inline lib/datasource/docker/index.ts:14]:
```suggestion
import { ensureTrailingSlash, trimTrailingSlash } from '../../util/url';
```

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/datasource/docker/index.ts (hunk 3 above)
Comment (candrews) [reply lib/datasource/docker/index.ts]: Thank you!

**Comment by candrews** [comment]
> @candrews Please do not force-push, simply merge master. We will squash the pr anyways. If you force-push we need to do a full review, because we don't see that's changed in latest commits

Noted - sorry about that!

**Comment by candrews** [comment]
@viceice @rarkins is this PR now in a mergable state?

**Review by viceice — APPROVED** [review]

**Pull request merged**